### ISSUE TITLE
Guide Structure Change: Sample Apps vs. Snippets

### ISSUE THREAD

**Issue by grayside** [GoogleCloudPlatform/samples-style-guide#105]
## Background

There are multiple types of samples. At a high level, there are narrowly focused snippets and deployable/executable sample applications. Most of the guidelines do not apply to sample applications.

We need a way to help folks the use the style guide understand:

1. What kind of sample are they creating/reviewing. We may keep the simplified set of 2, or we may want to go with a more granular list (e.g., snippet, pattern, app, notebook).
2. Which of the guidelines apply to them

## Proposed change

1. Add definitions of sample types to the style guide
2. Add a mechanism to separate guidance by type

For (2), there are two options:

1. Annotate each guideline for applicable sample types
2. Duplicate the guidelines per type

(1) is easier to maintain and harder to use
(2) is easier to evolve, easier to use, and harder to maintain

**Comment by grayside** [comment]
Here's an example of (1), which seems like a good incremental step:

![Screenshot 2023-04-19 at 9 46 28 AM](https://user-images.githubusercontent.com/283489/233144620-3608b8da-94bd-453c-9553-d3235aef2984.png)

**Comment by grayside** [comment]
Here's an example of "Duplicated":

```
.
├── _index.md
├── _apps.md
```

**Comment by grayside** [comment]
Here's an example of "Normalized":

```
.
├── _apps.md
├── _index.md
├── examples
│   ├── dotnet
│   ├── go
│   ├── java
│   ├── node
│   ├── php
│   ├── python
│   └── ruby
└── guidelines
    └── linting
```

**Comment by amanda-tarafa** [comment]
From the options you are proposing I like the annotated one better.

But I would like to propose another option, and that is to separate the guides in "code style guide" which will apply to all types of samples, and "sample guidelines", which I think will be different enough for different types of samples that it'd be OK to keep them separate.

Even the example you are using from annotated, i.e. Region Tags, I would argue is not the same for applications and snippets. In snippets, you want a region tag per snippet. In applications, I've seen plenty of time that there's a region tag per relevant part of the application (setup, action, error handling, cleanup), so that is very common that the same application has several region tags.

Basically, the burdens inferred from your table are very different depending on how similar are the guidelines for the different types of samples, if they are very similar the burden from duplication is great, but if they are almost entirely different then "duplication" is something we can live with.

**Comment by averikitsch** [comment]
I like annotated and normalize. Would it be possible to add a "page filter" based on the annotation in order to get the benefits of normalize but with the maintainability of annotated? 

I also like Amanda's proposal of a code style guide and sample guidelines, but that may be too big of a refactor.

Lastly, we might want to connect with iX to see if they have any learnings or best practices.

**Comment by grayside** [comment]
Here's a preview with some changes I made along the lines of what Amanda suggested: https://googlecloudplatform.github.io/samples-style-guide/previews/PR-129/
